src/containers/EditDonor/index.js

```diff
@@ -0,0 +1,221 @@
+      addrState: "",
+      zipCode: null,
+      institution: ""
+    }
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.getDonor(this.props.match.params.donorId).then(response => {
+      this.setState({
+        name: response.payload.donorName,
+        phone: response.payload.phone,
+        email: response.payload.email,
+        address: response.payload.address,
+        city: response.payload.city,
+        addrState: response.payload.state,
+        zipCode: response.payload.zipCode != null ? toString(response.payload.zipCode) : null,
+        institution: response.payload.institution
+      })
+    })
+  }
+
+
+
+  // validateForm() {
+  //   return this.email.length > 0 && this.password.length > 0;
+  // }
+
+
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { name, email, phone, address, addrState, city, zipCode, institution } = this.state;
+    const { updateDonor } = this.props;
+    updateDonor(this.props.match.params.donorId, name, email, phone, address, addrState, city, parseInt(zipCode), institution).then(response => {
+      this.setState({ submitted: true })
+      if (response.type === 'SUCCESS') {
+        this.setState({ success: true })
+      }
+      if (response.type === 'FAILURE') {
+        this.setState({ success: false })
+      }
+      setTimeout(() => {
+        this.setState({ submitted: false });
+      }, 3000);
+    });
+    window.scrollTo(0, this.myRef.current.top);
+  }
+
+  render() {
+    return (
+        <div>
+            <div ref={this.myRef}/>
+          <div style={{display: "flex", justifyContent: "center"}}>
+            {this.state.success && this.state.submitted ?
+                <Alert isOpen={this.state.visible} style={{width: "48rem"}} variant='success'> Successful donor
+                  update!</Alert>
+                : !this.state.success && this.state.submitted ?
+                    <Alert style={{width: "48rem"}} variant='danger'> Error!</Alert> : ''}
+          </div>
+            <div style={{display: "flex", justifyContent: "center"}}>
+              <Card style={{width: "48rem"}}>
+                <Card.Header as="h5">Edit Donor</Card.Header>
+                <Card.Body>
+                  <div className="Edit Donor">
+                    <form onSubmit={this.handleSubmit}>
+                      <Row>
+                        <Col>
+                          <label htmlFor="inputFullName">Full Name</label>
+                          {(this.state.name != null) ?
+                              <FormControl value={this.state.name}
+                                           onChange={e => this.setState({name: e.target.value})}
+                                           type="text"/>
+                              : <FormControl value={this.name}
+                                             onChange={e => this.setState({name: e.target.value})}
+                                             type="text"/>}
```

Comment:
I believe blocks like 
```
{(this.state.name != null) ?
                              <FormControl value={this.state.name}
                                           onChange={e => this.setState({name: e.target.value})}
                                           type="text"/>
                              : <FormControl value={this.name}
                                             onChange={e => this.setState({name: e.target.value})}
                                             type="text"/>}
```
can be simply replaced with:

 ```
<FormControl value={this.state.name || ''} 
                                           onChange={e => this.setState({name: e.target.value})}
                                           type="text"/>
```

Reply:
good call, much cleaner